pallets/democracy-mbm/src/lib.rs

```diff
@@ -127,24 +130,43 @@ impl<T: pallet_democracy::Config, W: weights::WeightInfo> SteppedMigration
                 break;
             }
 
+            let current_block_number =
+                frame_system::Pallet::<T>::block_number().saturated_into::<u32>();
```

Comment:
Isn't it still the same issue though?

E.g. if migration happens on block `N` and it takes 4 steps.
* 1st adjustment is done on block `N`
* 2nd adjustment is done on block `N+1`
* and so on

You should store the block number when the upgrade happens, and then reuse it for all steps.